### DISCUSSION TITLE
支持luarocks打包和库依赖管理

### DISCUSSION THREAD

**Discussion by 0xbillw** [cloudwu/skynet#1193] [Ideas]
发现Lua有个叫luarocks的打包和库依赖管理工具，可以对标Java的maven。
项目一多了会有代码复用，我们目前的解决办法：1. 手工拷贝文件； 2. 使用git submodule，很显然第1个最易出错也难维护，第2个倒是可行，只是更新时麻烦；很想有个类似Java maven的方案，luarocks应该可以。
请问云大大，skynet是否也可以使用luarocks呢，你们公司的lua项目是如何管理库依赖的？
谢谢！

**Comment by cloudwu** [comment]
我认为你可以先试用一下看是否合适。但我认为，skynet 有一个不太一样的生态。比如 luasocket 可能是 luarocks 生态中重要的一环，但你在 skynet 下是不可以用原生 socket  api 的。

另外，skynet 的不同项目，可能更强调服务的复用，而不是库的共用。如果不涉及 C 库，不同服务间依赖的基础库即使版本不同，问题也不是很大。因为 skynet 的不同服务是运行在独立 vm 中的，相互不会干扰。

所以，我建议一个完善的公共服务，可以连同它用到的基础库，使用 git submodule 引用即可。

如果有特别普适的基础库，也可以合在一起建一个子仓库。
